Disable default values

escopecz opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 4.x)
* a.b for any bug fixes (e.g. 4.0, 4.1, 4.2)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [y] <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | https://github.com/mautic/mautic-documentation/pull/321
| Related developer documentation PR URL | /
| Issue(s) addressed                     | /

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

#### Description:
This disables the default value for fields that Mautic uses to set someone as identified. This prevents the case where all visitors can be marked as identified if they have a default value set for first name. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Set a default value for first name
2. Visit a landing page or Mautic tracked page
3. Look for the contact in Mautic and notice they are not anonymous

#### Steps to test this PR:
1. Edit the custom field for first name and notice default value is disabled
2. Save and edit again and the default value should be empty
3. Edit email, company, first name, last name, any unique identifier field, or any social field and default value should be disabled. For company, companyname, companyemail, companywebsite, state, country and city should have default value disabled (all used to identify a company). 


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11250"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

alanhartless committed:
Disable default value for identifier fields

alanhartless committed:
Ensure defaultValue is set to empty in case an existing field already has something set

alanhartless committed:
Removed unused constant

alanhartless committed:
Allow entity to define default identifier fields through use of static method

escopecz committed:
The type is not always defined in the $data array. Check for that

mollux committed:
ensure the app folder is seen as the 4.3.x instead of a specific version (#11213)

mollux committed:
align lock file with change from  #11203 (#11218)

escopecz committed:
Fixes after cherry pick containing STAN, CS, libraries updated. It was 6yo code...

kuzmany reviewed (dismissed):
Works as expected

RCheesley reviewed (dismissed):
Was able to reproduce the issue with the current version of Mautic:
![Contacts _ Mautic (4)](https://user-images.githubusercontent.com/2930593/174846124-8cda1fd4-2616-47d6-b363-5eec3a70d9c3.png)

Confirm that the PR does prevent these fields from having a default value set.

We should make sure that we document which fields cannot have a default value set and why.

RCheesley commented:
Docs added at https://github.com/mautic/mautic-documentation/pull/321

escopecz committed:
Running `composer update --lock` after rebase

Package operations: 0 installs, 2 updates, 0 removals
  - Downloading guzzlehttp/psr7 (2.4.0)
  - Downloading guzzlehttp/guzzle (7.4.5)
  - Upgrading guzzlehttp/psr7 (2.3.0 => 2.4.0): Extracting archive
  - Upgrading guzzlehttp/guzzle (7.4.4 => 7.4.5): Extracting archive
